vllm/attention/backends/flash_attn.py

```diff
@@ -234,37 +255,35 @@ def forward(
                 output[:num_prefill_tokens] = out
             else:
                 # prefix-enabled attention
-                # TODO(Hai) this triton kernel has regression issue (broke) to
-                # deal with different data types between KV and FP8 KV cache,
-                # to be addressed separately.
-                output[:num_prefill_tokens] = PagedAttention.forward_prefix(
-                    query,
-                    key,
-                    value,
-                    key_cache,
-                    value_cache,
-                    prefill_meta.block_tables,
-                    prefill_meta.subquery_start_loc,
-                    prefill_meta.prompt_lens_tensor,
-                    prefill_meta.context_lens,
-                    prefill_meta.max_subquery_len,
-                    self.alibi_slopes,
+                # FIXME(woosuk): FlashAttention does not support FP8 KV cache.
+                output[:num_prefill_tokens] = flash_attn_varlen_func(
+                    q=query,
+                    k=key_cache,
+                    v=value_cache,
+                    cu_seqlens_q=prefill_meta.seq_start_loc,
```

Comment:
I think this needs to be 

                    prefill_meta.subquery_start_loc,
                    prefill_meta.seq_start_loc,
                    prefill_meta.max_subquery_len,
                    prefill_meta.max_prompt_len,

Reply:
https://github.com/vllm-project/vllm/pull/4560/commits/bf3abdc9e3cf044453ac5954f84c324f55c06501

Verified locally that this makes correctness test pass

Reply:
Thanks for bringing this up. I made the similar change along with some changes in Model Runner.